drivers: sensor: sdp3x: add driver

ygaillard opened the pull request:
This is a proposition for a driver using the SDP3x family differential pressure sensors. The related documentation can be found via https://sensirion.com/media/documents/4D045D69/6375F34F/DP_DS_SDP3x_digital_D1.pdf. The code was tested using a SDP31 chip on custom hardware in ambient conditions and return plausible data. 

Trigger and continous mode can be selected via project file. The board file select the different mesure mode provided by the sensor.

carlescufi commented:
Thanks for the PR! Please take a look at [our commit message guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines) to find out how to format your commit messages, and at [our contribution workflow](https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow) to understand how to update your Pull Request.

ygaillard commented:
The test stops and told I should contact my administrator.

ygaillard commented:
@carlescufi , finally fixed my issues.

MaureenHelm commented on drivers/sensor/sensirion/sdp3x/sdp3x.c:
```diff
@@ -0,0 +1,181 @@
+	const struct sdp3x_config *cfg = dev->config;
+	int rc = 0;
+
+	if (!device_is_ready(cfg->bus.bus)) {
```
Use `i2c_is_ready_dt`

ubieda replied:
Reference: https://github.com/zephyrproject-rtos/zephyr/blob/18fb3f9a329ade44495983206a5263f7aa2a84a4/drivers/mfd/mfd_max20335.c#L26

MaureenHelm commented on drivers/sensor/sensirion/sdp3x/sdp3x.h:
```diff
@@ -0,0 +1,46 @@
+
+#include <zephyr/device.h>
+
+#define SDP3X_CMD_READ_SERIAL	0x89
+#define SDP3X_CMD_RESET		0x0006
+
+#define SDP3X_RESET_WAIT_MS	25
+
+/*
+ * CRC parameters were taken from the
+ * "Checksum Calculation" section of the datasheet.
+ */
+#define SDP3X_CRC_POLY		0x31
+#define SDP3X_CRC_INIT		0xFF
+
+struct sdp3x_config {
+	struct i2c_dt_spec bus;
+	uint8_t mesure_mode;
+};
+
+struct sdp3x_data {
+	uint16_t t_sample;
+	uint16_t p_sample;
+};
+
+#ifdef CONFIG_SDP3X_PERIODIC_MODE
+static const uint16_t measure_cmd[4] = {
+	0x3603, 0x3608, 0x3615, 0x361E
+};
+#endif
+
+#ifdef CONFIG_SDP3X_SINGLE_SHOT_MODE
+static const uint16_t measure_cmd[4] = {
+	0x3624, 0x3726, 0x362F, 0x372D
+};
+#endif
```
Not worth having a separate private include file for this. Move the definitions to the .c file

ygaillard replied:
Change as suggested

MaureenHelm commented on dts/bindings/sensor/sensirion,sdp3x.yaml:
```diff
@@ -0,0 +1,26 @@
+include: [sensor-device.yaml, i2c-device.yaml]
+
+properties:
+  mesure_mode:
```
```suggestion
  measure-mode:
```

ygaillard replied:
Change as suggested

MaureenHelm reviewed (changes requested):
Thanks for the contribution!

Please add this sensor to `tests/drivers/build_all/sensor`

jeppenodgaard commented on drivers/sensor/sensirion/sdp3x/sdp3x.c:
```diff
@@ -0,0 +1,181 @@
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/__assert.h>
```
Does not seem to be used

ygaillard replied:
I agree for `kernel.h`and `asssert.h`. But remaining are necessary

ygaillard committed:
drivers: sensor: sdp3x: add driver

Add device driver for the SDP3x temperature and
differential pressure sensors.
Datasheet https://sensirion.com/media/documents/4D045D69/6375F34F/DP_DS_SDP3x_digital_D1.pdf
Signed-off-by: Yann Gaillard <yann@gaillard.onl>

ygaillard commented on drivers/sensor/sensirion/sdp3x/sdp3x.c:
```diff
@@ -0,0 +1,181 @@
+		return -ENOTSUP;
+	}
+
+	#ifdef CONFIG_SDP3X_SINGLE_SHOT_MODE
```
Change as sugegsted

ygaillard commented:
I implemented the changes. Did I miss something in the workflow?

kartben commented:
rebased to re-trigger stuck CI

ubieda commented on drivers/sensor/sensirion/sdp3x/sdp3x.c:
```diff
@@ -0,0 +1,213 @@
+#define SDP3X_CRC_INIT          0xFF
+
+struct sdp3x_config {
+        struct i2c_dt_spec bus;
```
Please fix indentation (various places): spaces -> tabs

ubieda reviewed (commented):
Thanks for your contribution! Code looks good but needs the indentation fixed and address Maureen's comment.

ygaillard committed:
Merge branch 'zephyrproject-rtos:main' into main
